### ISSUE TITLE
Data.copy() does not inherit units of individual axes

### ISSUE THREAD

**Issue by ddkohler** [wright-group/WrightTools#960]
```
d1 = wt.Data()
d1.create_variable(name="color", values=np.array([1]), units="nm")
d1.create_variable(name="length", values=np.array([1]), units="nm")
d1.transform("color", "length")

d1.color.convert("wn")
d2 = d1.copy()
print(d1.units, d2.units)  # d1 has units of (wn, nm) , d2 has units of (nm, nm)
```

This is problematic e.g. for controlling the axes units for `interact2D`, which copies the data object for plotting.

**Comment by darienmorrow** [comment]
This is a "gotcha" I have often encountered. 

`d1.color.convert("wn", convert_variables=True)` does what you intuitively want. 

I'm not sure what default behavior should be.

**Comment by darienmorrow** [comment]
Oops, upon testing, I don't think I am correct.

**Comment by ksunden** [comment]
This boils down to the attrs dictionary not being updated in `axes.convert`. this could be fixed by calling `self.parent._on_axes_updated()` in that method.

It can be worked around by calling `transform` _after_ the `convert` call

**Comment by ksunden** [comment]
(as `transform` and `Data.convert` both call `_on_axes_updated`)

**Comment by ksunden** [comment]
closed by #974
